standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -708,6 +762,24 @@ public void commitTxn(CommitTxnRequest rqst)
         lockInternal();
         dbConn = getDbConn(Connection.TRANSACTION_READ_COMMITTED);
         stmt = dbConn.createStatement();
+
+        if (rqst.isSetReplPolicy()) {
+          sourceTxnId = rqst.getTxnid();
```

Comment:
This code is duplicated in abortTxn as well. Can we have a private method, that returns target txn ID taking srcTxnId,replPolicy as input?